//
// PROJECT:         Aspia
// FILE:            protocol/system_info_session.proto
// LICENSE:         GNU General Public License 3
// PROGRAMMERS:     Dmitry Chapyshev (dmitry@aspia.ru)
//

syntax = "proto3";

option optimize_for = LITE_RUNTIME;

package aspia.proto.system_info;

enum Compressor
{
    COMPRESSOR_NONE = 0;
    COMPRESSOR_ZLIB = 1;
}

message CategoryListRequest
{
    uint32 dummy = 1;
}

message CategoryList
{
    repeated string uuid = 1;
}

message CategoryRequest
{
    string uuid  = 1;
    bytes params = 2;
}

message Category
{
    Compressor compressor = 1;
    bytes data            = 2;
}

message Request
{
    string request_uuid                       = 1;
    CategoryListRequest category_list_request = 2;
    CategoryRequest category_request          = 3;
}

message Reply
{
    string request_uuid         = 1;
    CategoryList category_list = 2;
    Category category          = 3;
}
